### PR TITLE
Fix enter key triggering cancel button on forms

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #11765 .
### Summary of Changes

Change button tag to a
### Testing Instructions

Create a new article
Add an article title
Click the alias field (or any text field), add text (or not).
Hit 'enter' key
#### Before patch

Article is cancelled, navigating back to article manager.
#### After patch

Nothing
### Documentation Changes Required

None

### CODE REVIEW

**Comment by 1apweb** [comment]
I have tested this item :white_check_mark: successfully on 295b12561e7ddebbb6f321eeaac34fb678f8d993<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11787">issues.joomla.org/joomla-cms/11787</a>.</sub>

**Comment by C-Lodder** [comment]
I have tested this item :white_check_mark: successfully on 295b12561e7ddebbb6f321eeaac34fb678f8d993<br /><br />Also tested in IE8 to see if it threw a hissy fit due to no `href` being used...all working and good to go.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11787">issues.joomla.org/joomla-cms/11787</a>.</sub>

**Comment by dgt41** [comment]
@C-Lodder do you think I should add `href="javascript:void(0)"` to be safe and fully compliant to standards?

File: administrator/components/com_categories/views/categories/tmpl/default_batch_footer.php [hunk 1]
```diff
@@ -9,9 +9,9 @@
 defined('_JEXEC') or die;
 
 ?>
-<button class="btn" type="button" onclick="document.getElementById('batch-category-id').value='';document.getElementById('batch-access').value='';document.getElementById('batch-language-id').value=''" data-dismiss="modal">
+<a class="btn" type="button" onclick="document.getElementById('batch-category-id').value='';document.getElementById('batch-access').value='';document.getElementById('batch-language-id').value=''" data-dismiss="modal">
```
Comment (yvesh) [inline administrator/components/com_categories/views/categories/tmpl/default_batch_footer.php]: As you already started cleaning that up, how about moving that to a javascript file or at least a script tag? Please? :)

That looks really..

**Comment by C-Lodder** [comment]
In HTML5 there's no need. If the `href` attribute isn't present, it acts as a placeholder.

File: administrator/components/com_categories/views/categories/tmpl/default_batch_footer.php (hunk 1 above)
Comment (dgt41) [inline administrator/components/com_categories/views/categories/tmpl/default_batch_footer.php]: I totally agree with you @yvesh, but not in this PR. After https://github.com/joomla/joomla-cms/pull/11654 gets merged, I will try to simplify, beautify all the modal related code

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on 295b12561e7ddebbb6f321eeaac34fb678f8d993<br /><br />thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11787">issues.joomla.org/joomla-cms/11787</a>.</sub>

**Comment by jeckodevelopment** [comment]
I have tested this item :white_check_mark: successfully on 295b12561e7ddebbb6f321eeaac34fb678f8d993<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11787">issues.joomla.org/joomla-cms/11787</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11787">issues.joomla.org/joomla-cms/11787</a>.</sub>

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on 295b12561e7ddebbb6f321eeaac34fb678f8d993<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11787">issues.joomla.org/joomla-cms/11787</a>.</sub>

**Pull request merged**